patterns/private-public-mirroring/README.md

```diff
@@ -0,0 +1,12 @@
+# Private-public mirroring
```

Comment:
Can we add a small section explaining why this pattern can be useful, and a small note saying that most of the time this pattern is seen for constant/immutable state variables?